tools/buildsys/src/manifest.rs

```diff
@@ -216,6 +244,33 @@ pub(crate) enum ImageFormat {
     Vmdk,
 }
 
+#[derive(Deserialize, Debug, Copy, Clone)]
+pub(crate) struct ImageLayout {
+    #[serde(default = "ImageLayout::default_os_image_size_gib")]
+    pub(crate) os_image_size_gib: u32,
+    #[serde(default = "ImageLayout::default_data_image_size_gib")]
+    pub(crate) data_image_size_gib: u32,
+}
+
+impl ImageLayout {
+    fn default_os_image_size_gib() -> u32 {
+        DEFAULT_OS_IMAGE_SIZE_GIB
+    }
+
+    fn default_data_image_size_gib() -> u32 {
+        DEFAULT_DATA_IMAGE_SIZE_GIB
+    }
+}
+
+impl Default for ImageLayout {
+    fn default() -> Self {
+        Self {
+            os_image_size_gib: Self::default_os_image_size_gib(),
```

Comment:
Out of curiosity - why not point these defaults directly at the constants?

Reply:
AFAIK that's not supported by `serde` per https://github.com/serde-rs/serde/issues/368.